### ISSUE TITLE
Salus Controls - it600 

### ISSUE THREAD

**Issue by rweise** [ioBroker/AdapterRequests#333]
Hallo,

eine Einbindung der Salus Systeme , 
für mich hauptsächlich das it600 System wäre wünschenswert.
Die Thermostate und Regler laufen über Zigbee, das wäre aber egal.
schöner wäre für mich zur Zeit ein Zugriff über die Salus-Control Server.

**Comment by DutchmanNL** [comment]
Gibs irgendwelche infos, links zur API zb?

**Comment by xyz4tech** [comment]
Maybe it‘s useful to have a look at the salus component for home assistant.

https://github.com/jvitkauskas/homeassistant_salus

**Comment by JeffJefferson** [comment]
Hi Leute,

ich habe mir mal die oben erwähnten sourcen angsehen. Scheinbar läuft ne REST API  auf dem UGE600. 

Leider finde ich die Readme nicht sehr aussagekräftig, also musste ich mich durch den code wühlen. Habe in der gateway.py einiges interessantes entdeckt. scheinbar muss die api kommunikation verschlüsselt werden, verstehe den teil in der encryption.py aber nicht so ganz. habe die api auch mal mit postman versucht anzusprechen, aber krieg da keine rückmeldung. (vermutlich wegen fehlender verschlüsselung)

habe das ganze daher einfach mal auf meinem rapsberry pi geschmissen, auf dem auch meine iobroker instanz läuft. der Aufruf der main.py  mittels ip und euid vom UGE600 sieht schonmal sehr gut aus.

Auf dem ersten Blick habe ich folgende json Endpunkte entdeckt:

 POST http://[ip vom uge 600]:80/deviceid/read 
und
POST http://[ip vom uge 600]:80/deviceid/write

in der main.py werden diese entpunkte mit unterschiedlichen json bodies bedient: 
bspw für den read endpunkt:
- **{"requestAttr": "readall"}**
- **{"requestAttr": "deviceid", "id": [{"data": {"DeviceType": 300, "UniID": "0000000000000000", "Endpoint": 0}}]}**

oder für den write entpunkt um die SOLL temperatur eines Thermostats auf 21°C festzulegen:
**{"requestAttr": "write", "id": [{"data": {"DeviceType": 100, "UniID": "001e5e090209fe13", "Endpoint": 9}, "sIT600TH": {"SetHeatingSetpoint_x100": 2100}}]}**

Leider weiß ich nicht wie man Adapter für Io broker erstellt, aber basierend auf dem sourcecode müsste das ja relativ einfach gehen. werden imerhin ja nur einzelne api's angesprochen. wäre auf jeden fall ganz cool wenn jemand so nen adapter basteln könnte :)

Liebe Grüße,
Jeff

**Comment by xyz4tech** [comment]
@DutchmanNL lässt sich mit den bisherigen Input etwas anfangen um eine Integration zu starten?

**Comment by mcm1957** [comment]
ioBroker und Python vertragen sich ziemlich schlecht. Die Einbindung in Installation von Python Modulen macht meist Probleme und sollte wenn irgend möglich vermniden werden,

**Comment by 2K-Darkness** [comment]
Ja, ein echter Adapter wäre mir natürlich lieber. Das es hier jetzt mit Python umgesetzt wurde ist natürlich auch kein "Muss".
Kein Plan wie aufwändig das wäre, die einfache Abfrage anders umzusetzen...

Aber ich kann natürlich gerne unterstützen.

Grüße :-)

**Comment by mcm1957** [comment]
Wenn du in die Adapterentwicklung einsteigen willst wirf einen Blick auf www.iobroker.dev. Und meld dich ggF in der Telegramm / Discord Starters Gruppe. Wenn du dich vor javascript nicht fürchtest wirst du dort sicher Hilfe bekommen mal was zu probieren.
